src/components/nav_drawer/nav_drawer.tsx

```diff
@@ -215,11 +283,11 @@ export class EuiNavDrawer extends Component {
     }
   };
 
-  modifyChildren = children => {
+  modifyChildren = (children: ReactElement[]): ReactElement[] => {
     // Loop through the EuiNavDrawer children (EuiListGroup, EuiHorizontalRules, etc)
     // Filter out falsy items
     const filteredChildren = React.Children.toArray(children);
-    return React.Children.map(filteredChildren, child => {
+    return React.Children.map(filteredChildren, (child: React.ReactElement) => {
```

Comment:
Likewise, this is processing nodes not elements. However, because of the prior `React.Children.toArray` this won't receive undefined or null values.

```suggestion
    return React.Children.map(filteredChildren, (child: NonNullable<ReactNode>) => {
```